File: 359-logger-rate-limiter/359-logger-rate-limiter.py
class Logger:

    def __init__(self):
        self.Q = deque()

    def shouldPrintMessage(self, timestamp: int, message: str) -> bool:

        while len(self.Q) > 0 and self.Q[0][0] <= timestamp - 10:
            self.Q.popleft()
        
        messages = [msg for time, msg in self.Q]
        
        if message in messages:
            return False
        
        self.Q.append([timestamp, message])
        return True


# Your Logger object will be instantiated and called as such:
# obj = Logger()
# param_1 = obj.shouldPrintMessage(timestamp,message)

File: 359-logger-rate-limiter/359-logger-rate-limiter.cpp
class Logger {
private:
    unordered_map<string, int> logTime;
public:
    Logger() {
        logTime = {};
    }
    
    bool shouldPrintMessage(int timestamp, string message) {
        if(logTime.count(message) == 0 or logTime[message]<= timestamp - 10){
            logTime[message] = timestamp;
            return true;
        } else {
            return false;
        }
    }
};

/**
 * Your Logger object will be instantiated and called as such:
 * Logger* obj = new Logger();
 * bool param_1 = obj->shouldPrintMessage(timestamp,message);
 */